src/instructlab/model/test.py

```diff
@@ -30,10 +36,102 @@
     default="auto",
     show_default=True,
 )
-@utils.macos_requirement(echo_func=click.secho, exit_exception=click.exceptions.Exit)
+# for Linux:
+@click.option(
+    "-m",
+    "--model",
+    default=cfg.DEFAULT_MODEL,
+    show_default=True,
+    help="Base model name to test on Linux",
+)
+@click.option(
+    "-t",
+    "--test_file",
+    help="Test data file",
+)
+@click.option(
+    "--endpoint-url",
+    type=click.STRING,
+    help="Custom URL endpoint for OpenAI-compatible API. Defaults to the `ilab model serve` endpoint.",
+)
+@click.option(
+    "--api-key",
+    type=click.STRING,
+    default=cfg.DEFAULT_API_KEY,  # Note: do not expose default API key
+    help="API key for API endpoint. [default: cfg.DEFAULT_API_KEY]",
+)
+@click.option(
+    "--model-family",
+    help="Force model family to use when picking a generation template",
+)
+@click.pass_context
 # pylint: disable=function-redefined
-def test(data_dir, model_dir, adapter_file):
+def test(
+    ctx,
+    data_dir,
+    # for macOS:
+    model_dir,
+    adapter_file,
+    # for Linux:
+    model,
+    test_file,
+    endpoint_url,
+    api_key,
+    model_family,
+):
     """Runs basic test to ensure model correctness"""
+
+    if not utils.is_macos_with_m_chip():
+        ctx.obj.logger.debug("")
+        # pylint: disable=import-outside-toplevel
+        # Third Party
+        import httpx
+
+        # First Party
+        from instructlab.server import ensure_server
+
+        # Local
+        from .linux_test import linux_test
+
+        server_process = server_queue = None
+        if not endpoint_url:
+            try:
+                server_process, endpoint_url, server_queue = ensure_server(
+                    ctx.obj.logger,
+                    ctx.obj.config.serve,
+                    True,
+                    "",
+                    "",
+                    "",
+                    model_family,
+                )
+            except Exception as exc:
+                click.secho(f"Failed to start server: {exc}", fg="red")
+                raise click.exceptions.Exit(1)
+        if not endpoint_url:
+            endpoint_url = ctx.obj.config.serve.api_base()
+        ctx.obj.logger.debug("%s", endpoint_url)
+        if not test_file:
+            test_file = sorted(
+                Path(ctx.obj.config.generate.output_dir).glob("test_*"), reverse=True
+            )[0]
+        ctx.obj.logger.debug("test_file=%s", test_file)
+        linux_test(
+            ctx,
+            endpoint_url,
+            test_file,
+            api_key=api_key,
+            models=[model, Path("models") / "ggml-model-f16.gguf"],
+            create_params={"max_tokens": 100},
+        )
+        if server_process and server_queue:
+            server_process.terminate()
+            server_process.join(timeout=30)
+            server_queue.close()
+            server_queue.join_thread()
+
+        return
```

Comment:
This is arguable, doing `return` in the middle of the function to separate two conditions is a bit unusual, isn't it?
Why not `if not macos: ... else ... fi`? To avoid the re-indent?

Reply:
I know `fi` in shell.  What is `fi` here?
I suppose the rest of code should be refactored anyway to separate user commands and platform dependent code.

Reply:
Yes sorry, it's a typo the `fi` does not exist in Python but you get the idea, right?

Reply:
I agree that it looks asymmetric and it is not good. But I haven't got idea how to avoid the re-indent.

Reply:
If we have two functions, we can have a smaller re-indent (but code move). It this provides clarify I don't mind the re-indent.